Assets/2d_TDP/Script/DialogBubble.cs

```diff
@@ -0,0 +1,164 @@
+				}
+				else 
+				{
+					//create bubble
+					vBubbleObject = Instantiate(Resources.Load<GameObject> ("Customs/BubbleRound"));
+					vBubbleObject.transform.position = vcharacter.transform.position + new Vector3(0.15f, 1.75f, 0f); //move a little bit the teleport particle effect
+				}
+
+				//show the mouse and wait for the user to left click OR NOT (if not, after 10 sec, it disappear)
+				vBubbleObject.GetComponent<Appear>().needtoclick = vBubble.vClickToCloseBubble;
+				
+				Color vNewBodyColor = new Color(vBubble.vBodyColor.r, vBubble.vBodyColor.g, vBubble.vBodyColor.b, 0f);
+				Color vNewBorderColor = new Color(vBubble.vBorderColor.r, vBubble.vBorderColor.g, vBubble.vBorderColor.b, 0f);
+				Color vNewFontColor = new Color(vBubble.vFontColor.r, vBubble.vFontColor.g, vBubble.vFontColor.b, 255f);
+				
+				//get all image below the main Object
+				foreach (Transform child in vBubbleObject.transform)
+				{
+					SpriteRenderer vRenderer = child.GetComponent<SpriteRenderer> ();
+					TextMesh vTextMesh = child.GetComponent<TextMesh> ();
+					
+					if (vRenderer != null && child.name.Contains("Body"))
+					{
+						//change the body color
+						vRenderer.color = vNewBodyColor;
+						
+						if (vRenderer.sortingOrder < 10)
+							vRenderer.sortingOrder = 1500;
+					}
+					else if (vRenderer != null && child.name.Contains("Border"))
+					{
+						//change the border color
+						vRenderer.color = vNewBorderColor;
+						if (vRenderer.sortingOrder < 10)
+							vRenderer.sortingOrder = 1501;
+					} 
+					else if (vTextMesh != null && child.name.Contains("Message"))
+					{
+						//change the message and show it in front of everything
+						vTextMesh.color = vNewFontColor;
+						vTextMesh.text = vTrueMessage;
+						child.GetComponent<MeshRenderer>().sortingOrder = 1550;
+						
+						Transform vMouseIcon = child.Find("MouseIcon");
+						if (vMouseIcon != null && !vBubble.vClickToCloseBubble)
+							vMouseIcon.gameObject.SetActive(false);
+					}
+					
+					//disable the mouse icon because it will close by itself
+					if (child.name == "MouseIcon" && !vBubble.vClickToCloseBubble)
+						child.gameObject.SetActive(false);
+					else
+						vActiveBubble =  vBubble; //keep the active bubble and wait for the Left Click
+				}
+				
+				vcharacter.vCurrentBubble = vBubbleObject; //attach it to the player
+				vBubbleObject.transform.parent = vcharacter.transform; //make him his parent
+			} else if (vActiveBubble == vBubble && vActiveBubble.vClickToCloseBubble)
+			{
+				gotonextbubble = true;
+				vcharacter.vCurrentBubble = null;
+			}
+		}
+	}	
+
+	void Update () 
+	{
+		//check if we have the mouse over the character
+		ray = Camera.main.ScreenPointToRay (Input.mousePosition);
+	
+		//make sure we left click and is on a NPC
+		if (Physics.Raycast (ray, out hit) && Input.GetMouseButtonDown (0)) {
```

Comment:
Are we introducing clicking to the game? I was under the impression that using F for interaction was going to be our method of choice. Of course, we should set something up like prioritizing what you're looking at for interaction

Reply:
Also this is part of script from the text bubble lib. I thought we are using mouse to navigate the camera around the map, so maybe clicking for text (not for control) could make some sense.

Reply:
that could make some sense. I would say let's adapt this to be triggerable by either for now, and we can make a decision on whether clicking should also work later. The mouse seems like it'll be a less important tool than F for us, so we can later decide how important it should be. @zw1215